Validation/HGCalValidation/plugins/HGCalWaferValidation.cc

```diff
@@ -213,36 +262,44 @@ void HGCalWaferValidation::ProcessWaferLayer(DDCompactView::GraphWalker& walker)
       //edm::LogVerbatim(logcat) << "rotStr " << rotStr << " rotCode " << rotCode;
 
       // convert shape code to wafer types defined in HGCalTypes.h
-      waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferOut;
-      if (shapeStr == "F")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFull;
-      else if (shapeStr == "a")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferHalf;
-      else if (shapeStr == "am")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferHalf2;
-      else if (shapeStr == "b")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFive;
-      else if (shapeStr == "bm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferFive2;
-      else if (shapeStr == "c")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferThree;
-      else if (shapeStr == "d")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferSemi;
-      else if (shapeStr == "dm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferSemi2;
-      else if (shapeStr == "g")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferChopTwo;
-      else if (shapeStr == "gm")
-        waferInfo.shapeCode = HGCalTypes::WaferPartialType::WaferChopTwoM;
+      waferInfo.shapeCode = waferShapeMapDD.at(shapeStr);
```

Comment:
`at` has `bounds-checking` and, therefore, is slower. Since this is not reconstruction code, we can skip that for now and eventually use the direct `[]` access operator for later.

Reply:
Yes, I did try to use `[]` first but failed to compile because it cannot accept `const std::map`.